### PR TITLE
feat(design): daff-feature component

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/design/src/molecules/feature/feature-body/feature-body.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { DaffFeatureBodyComponent } from './feature-body.component';
+
+describe('DaffFeatureBodyComponent', () => {
+  it('should create an instance', () => {
+    const component = new DaffFeatureBodyComponent();
+    expect(component).toBeTruthy();
+  });
```
Comment (damienwebdev) [inline libs/design/src/molecules/feature/feature-body/feature-body.component.spec.ts]: Needs a test.

File: libs/design/src/molecules/feature/feature-title/feature-title.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { DaffFeatureTitleComponent } from './feature-title.component';
+
+describe('DaffFeatureTitleComponent', () => {
+  it('should create an instance', () => {
+    const component = new DaffFeatureTitleComponent();
+    expect(component).toBeTruthy();
+  });
```
Comment (damienwebdev) [inline libs/design/src/molecules/feature/feature-title/feature-title.component.spec.ts]: Needs a test.

File: libs/design/src/molecules/feature/feature-body/feature-body.component.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: '[daff-feature-body]',
+  template: '<ng-content></ng-content>',
+  host: {'class': 'daff-feature__body'}
+})
```
Comment (damienwebdev) [inline libs/design/src/molecules/feature/feature-body/feature-body.component.ts]: `changeDetection: ChangeDetectionStrategy.OnPush`

File: libs/design/src/molecules/feature/feature-icon/feature-icon.component.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Component } from '@angular/core';
+
+@Component ({
+  selector: '[daff-feature-icon]',
+  template: '<ng-content></ng-content>',
+  host: {'class': 'daff-feature__icon'}
```
Comment (damienwebdev) [inline libs/design/src/molecules/feature/feature-icon/feature-icon.component.ts]: changeDetection: ChangeDetectionStrategy.OnPush

File: libs/design/src/molecules/feature/feature-title/feature-title.component.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: '[daff-feature-title]',
+  template: '<ng-content></ng-content>',
+  host: {'class': 'daff-feature__title'}
```
Comment (damienwebdev) [inline libs/design/src/molecules/feature/feature-title/feature-title.component.ts]: changeDetection: ChangeDetectionStrategy.OnPush

File: libs/design/src/molecules/feature/feature.component.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { Component, ViewEncapsulation } from '@angular/core';
+
+@Component({
+  selector: 'daff-feature',
+  templateUrl: './feature.component.html',
+  styleUrls: ['./feature.component.scss'],
+  host: {'class': 'daff-feature'},
+  encapsulation: ViewEncapsulation.None,
```
Comment (damienwebdev) [inline libs/design/src/molecules/feature/feature.component.ts]: changeDetection: ChangeDetectionStrategy.OnPush

File: libs/design/src/molecules/feature/feature.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(DaffFeatureComponent);
+    component = fixture.componentInstance;
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
Comment (damienwebdev) [inline libs/design/src/molecules/feature/feature.component.spec.ts]: Needs a test about adding a class to host.

File: libs/design/src/molecules/feature/feature.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+### Icon
+* Feature icon is used by adding `<daff-feature-icon>` to any tag.
+
+### Usage
+```
+<daff-feature>
+  <img daff-feature-icon src="" />
+  <h4 daff-feature-title>Title</h4>
+  <p daff-feature-body>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
```
Comment (damienwebdev) [inline libs/design/src/molecules/feature/feature.md]: Include "some other content" as if a user wanted to add additional content.

File: libs/design/src/molecules/feature/feature-body/feature-body.component.spec.ts (hunk 1 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature-body/feature-body.component.spec.ts]: Resolved in 688add3

File: libs/design/src/molecules/feature/feature-title/feature-title.component.spec.ts (hunk 2 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature-title/feature-title.component.spec.ts]: Resolved in 688add3

File: libs/design/src/molecules/feature/feature-body/feature-body.component.ts (hunk 3 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature-body/feature-body.component.ts]: Resolved in 688add3

File: libs/design/src/molecules/feature/feature-icon/feature-icon.component.ts (hunk 4 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature-icon/feature-icon.component.ts]: Resolved in 688add3

File: libs/design/src/molecules/feature/feature-title/feature-title.component.ts (hunk 5 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature-title/feature-title.component.ts]: Resolved in 688add3

File: libs/design/src/molecules/feature/feature.component.ts (hunk 6 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature.component.ts]: Resolved in 688add3

File: libs/design/src/molecules/feature/feature.md (hunk 8 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature.md]: Resolved in 2232863

File: libs/design/src/molecules/feature/feature.component.spec.ts (hunk 7 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature.component.spec.ts]: Resolved in 41f8ddf

File: libs/design/src/molecules/feature/feature.component.ts (hunk 6 above)
Comment (xelaint) [reply libs/design/src/molecules/feature/feature.component.ts]: Resolved in 41f8ddf

**Pull request merged**